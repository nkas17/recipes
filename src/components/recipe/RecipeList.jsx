import React from 'react';
import PropTypes from 'prop-types';
import RecipeListRow from './RecipeListRow';

/**
 * List of recipes
 */
const RecipeList = ({ recipes, deleteRecipe, showDelete }) => (
	<table className="table">
		<tbody>
			{recipes.map(recipe => (
				<RecipeListRow
					key={recipe.id}
					recipe={recipe}
					deleteRecipe={deleteRecipe}
					showDelete={showDelete}
				/>
			))}
		</tbody>
	</table>
);

RecipeList.propTypes = {
	/**
	 * list of recipes to display
	 */
	recipes: PropTypes.arrayOf(PropTypes.object).isRequired,

	/**
	 * function to delete a recipe
	 */
	deleteRecipe: PropTypes.func.isRequired,

	/**
	 * Show delete button or not.
	 */
	showDelete: PropTypes.bool.isRequired,
};

export default RecipeList;
