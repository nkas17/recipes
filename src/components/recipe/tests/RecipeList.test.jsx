import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import RecipeList from '../RecipeList';

const recipes = [
	{
		id: 'tacos',
		title: 'tacos',
		category: 'other',
		description: 'Tasty homemade tacos - hard or soft',
		ingredients: ['cheese', 'ground beef', 'tortilla'],
	},
	{
		id: 'quiche',
		title: 'quiche',
		category: 'other',
		description: 'Delicious Spinach Quiche and Ham & Cauliflower quiche',
		ingredients: ['cheese', 'pie-crust', 'egg'],
	},
];

describe('RecipeList', () => {
	it('renders correctly', () => {
		const tree = shallow(
			<RecipeList recipes={recipes} showDelete deleteRecipe={() => {}} />
		);
		expect(toJSON(tree)).toMatchSnapshot();
	});
});
