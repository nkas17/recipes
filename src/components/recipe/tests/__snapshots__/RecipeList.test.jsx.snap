// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList renders correctly 1`] = `
<table
  className="table"
>
  <tbody>
    <RecipeListRow
      deleteRecipe={[Function]}
      key="tacos"
      recipe={
        Object {
          "category": "other",
          "description": "Tasty homemade tacos - hard or soft",
          "id": "tacos",
          "ingredients": Array [
            "cheese",
            "ground beef",
            "tortilla",
          ],
          "title": "tacos",
        }
      }
      showDelete={true}
    />
    <RecipeListRow
      deleteRecipe={[Function]}
      key="quiche"
      recipe={
        Object {
          "category": "other",
          "description": "Delicious Spinach Quiche and Ham & Cauliflower quiche",
          "id": "quiche",
          "ingredients": Array [
            "cheese",
            "pie-crust",
            "egg",
          ],
          "title": "quiche",
        }
      }
      showDelete={true}
    />
  </tbody>
</table>
`;
